src/unaudited/ECDSAStakeRegistry.sol

```diff
@@ -0,0 +1,385 @@
+        (address[] memory signers, bytes[] memory signatures, uint32 referenceBlock) = abi.decode(
+            _signatureData,
+            (address[], bytes[], uint32)
+        );
+        _checkSignatures(_dataHash, signers, signatures, referenceBlock);
+        return IERC1271Upgradeable.isValidSignature.selector;
+    }
+
+    /// @notice Retrieves the current stake quorum details.
+    /// @return Quorum - The current quorum of strategies and weights
+    function quorum() external view returns (Quorum memory) {
+        return _quorum;
+    }
+
+    /// @notice Retrieves the last recorded weight for a given operator.
+    /// @param _operator The address of the operator.
+    /// @return uint256 - The latest weight of the operator.
+    function getLastCheckpointOperatorWeight(address _operator) external view returns (uint256) {
+        return _operatorWeightHistory[_operator].latest();
+    }
+
+    /// @notice Retrieves the last recorded total weight across all operators.
+    /// @return uint256 - The latest total weight.
+    function getLastCheckpointTotalWeight() external view returns (uint256) {
+        return _totalWeightHistory.latest();
+    }
+
+    /// @notice Retrieves the operator's weight at a specific block number.
+    /// @param _operator The address of the operator.
+    /// @param _blockNumber The block number to get the operator weight for the quorum
+    /// @return uint256 - The weight of the operator at the given block.
+    function getOperatorWeightAtBlock(
+        address _operator,
+        uint32 _blockNumber
+    ) external view returns (uint256) {
+        return _operatorWeightHistory[_operator].getAtBlock(_blockNumber);
+    }
+
+    /// @notice Retrieves the total weight at a specific block number.
+    /// @param _blockNumber The block number to get the total weight for the quorum
+    /// @return uint256 - The total weight at the given block.
+    function getLastCheckpointTotalWeightAtBlock(
+        uint32 _blockNumber
+    ) external view returns (uint256) {
+        return _totalWeightHistory.getAtBlock(_blockNumber);
+    }
+
+    function operatorRegistered(address _operator) external view returns (bool) {
+        return _operatorRegistered[_operator];
+    }
+
+    /// @notice Calculates the current weight of an operator based on their delegated stake in the strategies considered in the quorum
+    /// @param _operator The address of the operator.
+    /// @return uint256 - The current weight of the operator; returns 0 if below the threshold.
+    function getOperatorWeight(address _operator) public view returns (uint256) {
+        StrategyParams[] memory strategies = _quorum.strategies;
+        uint256 weight;
+        uint256 sharesAmount;
+        for (uint256 i; i < strategies.length; i++) {
+            sharesAmount = DELEGATION_MANAGER.operatorShares(_operator, strategies[i].strategy);
+            weight += sharesAmount * strategies[i].multiplier;
+        }
```

Comment:
there's now an arrayified version of this function which should be more efficient to call https://github.com/Layr-Labs/eigenlayer-contracts/blob/dev/src/contracts/core/DelegationManager.sol#L864-L873